test/e2e/node_problem_detector.go

```diff
@@ -149,7 +149,7 @@ var _ = framework.KubeDescribe("NodeProblemDetector", func() {
 				"involvedObject.name":      node.Name,
 				"involvedObject.namespace": v1.NamespaceAll,
 				"source":                   source,
-			}.AsSelector()
+			}.AsSelector().String()
```

Comment:
Not having any place to store the compiled field selector in v1 makes this uglier than it should be.  We don't need to change it here, but I hit this when moving ListOptions to meta.k8s.io/v1, and felt that it should easier / possible to deal in v1 without this.  Not sure what the best approach is though.